notifications_utils/broadcast_areas/__init__.py

```diff
@@ -63,98 +49,94 @@ def __init__(self, feature):
     def __eq__(self, other):
         return self.id == other.id
 
-    @property
-    def name(self):
-        for possible_name_key in {
-            'rgn18nm', 'ctyua16nm', 'ctry19nm',
-        }:
-            with suppress(KeyError):
-                return self.feature['properties'][possible_name_key]
-
-        raise KeyError(f'No name found in {self.feature["properties"]}')
-
-    @property
-    def polygons(self):
-        if self.feature['geometry']['type'] == 'MultiPolygon':
+    def _polygons(self, feature):
+        if feature['geometry']['type'] == 'MultiPolygon':
             return [
                 polygons[0]
-                for polygons in self.feature['geometry']['coordinates']
+                for polygons in feature['geometry']['coordinates']
             ]
-        if self.feature['geometry']['type'] == 'Polygon':
+        if feature['geometry']['type'] == 'Polygon':
             return [
-                self.feature['geometry']['coordinates'][0]
+                feature['geometry']['coordinates'][0]
             ]
         raise TypeError(
             f'Unknown geometry type {self.feature["geometry"]["type"]} '
             f'in {self.__class__.__name} {self.name}'
         )
 
-    @property
-    def unenclosed_polygons(self):
+    def _unenclosed_polygons(self, feature):
         # Some mapping tools require shapes to be unenclosed, i.e. the
         # last point joins the first point implicitly
         return [
-            coordinates[:-1] for coordinates in self.polygons
+            coordinates[:-1] for coordinates in self._polygons(feature)
         ]
 
+    @property
+    def polygons(self):
+        return self._polygons(self.feature)
 
-class BroadcastAreaLibrary(SerialisedModelCollection, IdFromNameMixin, GetItemByIdMixin):
+    @property
+    def unenclosed_polygons(self):
+        return self._unenclosed_polygons(self.feature)
 
-    model = BroadcastArea
+    @property
+    def simple_polygons(self):
+        return self._polygons(self.simple_feature)
 
-    def __init__(self, filename):
-        self.name, geojson_data = load_geojson_file(filename)
-        self.items = geojson_data['features']
+    @property
+    def simple_unenclosed_polygons(self):
+        return self._unenclosed_polygons(self.simple_feature)
 
-    def get_examples(self, max_displayed=4):
+    @cached_property
+    def feature(self):
+        return geojson.loads(self._feature)
 
-        truncate_at = max_displayed - 1
+    @cached_property
+    def simple_feature(self):
+        return geojson.loads(self._simple_feature)
 
-        names = [area.name for area in sorted(self)]
-        count_of_excess_names = len(names) - truncate_at
+    @property
+    def sub_areas(self):
+        return [
+            BroadcastArea(row)
+            for row in BroadcastAreasRepository().get_all_areas_for_group(self.id)
```

Comment:
Is there potentially some inefficiency in instantiating and connecting to the database each time we call `.sub_areas` (and others)? Would it not be better to do this once at the time of starting the app?